library/general/src/modules/PackagesProposal.rb

```diff
@@ -248,15 +253,21 @@ def RemoveTaboos(unique_id, resolvables)
     #
     # @param [String] unique_id the unique identificator
     # @return [Array<String>] List of taboos for the given ID
-    def GetTaboos(unique_id)
-      @taboos[unique_id] || []
+    def GetTaboos(unique_id, type)
+      return [] unless @taboos.key?(unique_id) && @taboos[unique_id].key?(type)
+
+      @taboos[unique_id][type]
     end
 
     # Returns the full list of taboos
     #
     # @return [Array<String>] List of taboos
-    def GetAllTaboos
-      @taboos.values.flatten.uniq
+    def GetAllTaboos(type)
+      all_taboos = @taboos.values.each_with_object([]) do |tab, all|
+        all.concat(tab[type]) if tab.key?(type)
```

Comment:
can this potentially also add nil? if so, then compact is missing.

Reply:
I do not get how it could add nil. We are checking `tab.key?(type)` and, in that case, we concatenate `tab[type]` which is expected to be an array.